frontend/src/test/scala/bloop/SourcesGlobsCompileSpec.scala

```diff
@@ -0,0 +1,68 @@
+package bloop
+
+import bloop.logging.RecordingLogger
+import scala.concurrent.Promise
+import bloop.util.TestUtil
+import java.nio.file.Files
+import scala.concurrent.Await
+import scala.concurrent.duration.FiniteDuration
+import bloop.util.TestProject
+import java.util.concurrent.TimeUnit
+import bloop.cli.ExitStatus
+import bloop.config.Config
+import bloop.data.SourcesGlobs
+
+object SourcesGlobsCompileSpec extends bloop.testing.BaseSuite {
+
+  test("compile respects sources globs field") {
+    TestUtil.withinWorkspace { workspace =>
+      val sources = List(
+        """/main/scala/Foo.scala
+          |package foo
+          |class Foo
+          """.stripMargin,
+        """/main/scala/FooTest.scala
+          |package foo
+          |class Foo // would be compile error if this was not excluded via globs.
+          """.stripMargin
+      )
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val baseProject = TestProject(workspace, "a", sources)
+      val globDirectory =
+        baseProject.config.directory.resolve("src").resolve("main").resolve("scala")
+      val `A` = baseProject.copy(
+        config = baseProject.config.copy(
+          sources = Nil,
+          sourcesGlobs = Some(
+            List(
+              Config.SourcesGlobs(
+                globDirectory,
+                walkDepth = None,
+                includes = List("glob:*.scala"),
+                excludes = List("glob:*Test.scala")
+              )
+            )
+          )
+        )
+      )
+      val projects = List(`A`)
+      val state = loadState(workspace, projects, logger)
+      val compiledState = state.compile(`A`)
+      assertExitStatus(compiledState, ExitStatus.Ok)
```

Comment:
Should we use `assertValidCompilationState` to make sure that the compilation class files and certain file system invariants hold after compilation?

Reply:
Good idea. Done.